internal/nodes/ingress_model_rel.go

```diff
@@ -622,6 +652,33 @@ func validateGatewayForClass(key string, gateway *advl4v1alpha1pre1.Gateway) err
 	return nil
 }
 
+func validateSvcApiGatewayForClass(key string, gateway *svcapiv1alpha1.Gateway) error {
+	gwClassObj, err := lib.GetSvcAPIInformers().GatewayClassInformer.Lister().Get(gateway.Spec.GatewayClassName)
+	if err != nil {
+		utils.AviLog.Errorf("key: %s, msg: Unable to fetch corresponding networking.x-k8s.io/gatewayclass %s %v",
+			key, gateway.Spec.GatewayClassName, err)
+		return err
+	}
+
+	for _, listener := range gateway.Spec.Listeners {
+		gwName, nameOk := listener.Routes.Selector.MatchLabels[lib.GatewayNameLabelKey]
+		gwNamespace, nsOk := listener.Routes.Selector.MatchLabels[lib.GatewayNamespaceLabelKey]
+		if !nameOk || !nsOk ||
+			(nameOk && gwName != gateway.Name) ||
+			(nsOk && gwNamespace != gateway.Namespace) {
+			return errors.New("Incorrect gateway matchLabels configuration")
```

Comment:
For a call from isGatewayDelete(), even if one listener has a mismatch, should we delete the gateway ?

Reply:
I think so - it's a broken configuration.